llvm/test/SYCLLowerIR/ESIMD/lower_global_stores.ll

```diff
@@ -0,0 +1,24 @@
+; This test checks whether global stores are converted to vstores
+;
+; RUN: opt < %s -passes=LowerESIMD -S | FileCheck %s                            
+
+target datalayout = "e-i64:64-v16:16-v24:32-v32:32-v48:64-v96:128-v192:256-v256:256-v512:512-v1024:1024-n8:16:32:64"
+target triple = "spir64-unknown-unknown"
+
+%"class.sycl::_V1::ext::intel::esimd::simd" = type { %"class.sycl::_V1::ext::intel::esimd::detail::simd_obj_impl" }
+%"class.sycl::_V1::ext::intel::esimd::detail::simd_obj_impl" = type { <16 x float> }
+
+@va = dso_local global %"class.sycl::_V1::ext::intel::esimd::simd" zeroinitializer, align 64 #0
+@vb = dso_local global %"class.sycl::_V1::ext::intel::esimd::simd" zeroinitializer, align 64 #0
+
+define weak_odr dso_local spir_kernel void @foo() {
+%1 = call <16 x float> asm "", "=rw"()
+; CHECK: call void @llvm.genx.vstore.v16f32.p0v16f32(<16 x float> %1, <16 x float>* getelementptr inbounds (%"class.sycl::_V1::ext::intel::esimd::simd", %"class.sycl::_V1::ext::intel::esimd::simd"* @va, i64 0, i32 0, i32 0))
```

Comment:
If this approach works, then it is Ok with me.
My biggest concern though is that we are supposed to start replacing GenX intrinsic calls with LLVM IR and SPIRV extensions, but here is go backward - from LLVM IR to GenX call.

@vmustya - can you please advise if there is some better way than using getn.vstore here?

Reply:
Yeah this is a good point, I don't know how we can represent this semantic in native SPIR-V.

I think it's somewhat tangential to this PR since we already manually insert vstores in a ton of places in headers, so it's not like we are started using a genx intrinsic we never used before. However if there's an easy future-proof fix I'm happy to implement it.

Reply:
I've got response in IM that LLVM IR is not usable, and vstore is the only option for now. Perhaps, will be changed soon and in separate PRs. Thank you.

Reply:
cool thanks